lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/AzureBatchResourceManager.java

```diff
@@ -46,6 +47,7 @@
 
   private final Map<String, ResourceRequestEvent> containerRequests;
 
+  private final ArrayList<String> activeContainerIds;
```

Comment:
Correct me if I am wrong, this activeContainerIds is quite overlapped with containerRequests in term of the usage. The only difference I see is  : containerRequests contains requested resource, but activeContainerIds contains launched resource - they both are removed when they are released. 

Is there a case that resource request got rejected? If we don't have it now,  Can we merge these two caches into one? 
Or I don't see the logic we could handle rejected request.

Maintaining cache is always painful.

Reply:
We want to schedule the timer if there is a containerRequest (i.e. a container was requested).
However, if we trigger the onResourceStatus() on REEFEventHandlers when there is no container that is launched, an exception is raised. Hence the need to track these two separately.